packages/dropdowns/src/styled/menu/StyledMenu.tsx

```diff
@@ -6,15 +6,57 @@
  */
 
 import React, { HTMLAttributes } from 'react';
-import styled from 'styled-components';
-import classNames from 'classnames';
-import { retrieveComponentStyles, isRtl, DEFAULT_THEME } from '@zendeskgarden/react-theming';
-import MenuStyles from '@zendeskgarden/css-menus';
-import ArrowStyles from '@zendeskgarden/css-arrows';
-import { POPPER_PLACEMENT } from '../../utils/garden-placements';
+import styled, { keyframes, css, ThemeProps, DefaultTheme } from 'styled-components';
+import {
+  retrieveComponentStyles,
+  DEFAULT_THEME,
+  getColor,
+  arrowStyles
+} from '@zendeskgarden/react-theming';
+import { POPPER_PLACEMENT, getArrowPosition } from '../../utils/garden-placements';
 
 const COMPONENT_ID = 'dropdowns.menu';
 
+const ANIMATE_UP_KEYFRAME = keyframes`
+  0% {
+    margin-bottom: -${DEFAULT_THEME.space.base * 5}px;
+  }
+
+  100% {
+    margin-bottom: 0;
+  }
+`;
+
+const ANIMATE_RIGHT_KEYFRAME = keyframes`
+  0% {
+    margin-left: -${DEFAULT_THEME.space.base * 5}px;
+  }
+
+  100% {
+    margin-left: 0;
+  }
+`;
+
+const ANIMATE_DOWN_KEYFRAME = keyframes`
+  0% {
+    margin-top: -${DEFAULT_THEME.space.base * 5}px;
+  }
+
+  100% {
+    margin-top: 0;
+  }
+`;
+
+const ANIMATE_LEFT_KEYFRAME = keyframes`
+  0% {
+    margin-right: -${DEFAULT_THEME.space.base}px;
+  }
+
+  100% {
+    margin-right: 0;
+  }
+`;
```

Comment:
I wasn't sure if this would have a negative performance impact. Looks like this doesn't affect it was long as the positions don't change rapidly. I've updated this to be instantiated inline.